app/models/ems_refresh/save_inventory.rb

```diff
@@ -89,7 +91,7 @@ def save_vms_inventory(ems, hashes, target = nil, disconnect = true)
 
           # Find the Vm in the database with the current uid_ems.  In the event
           #   of duplicates, try to determine which one is correct.
-          found = vms_by_uid_ems[h[:uid_ems]] || []
+          found = Array(vms_by_ems_ref[h[:ems_ref]] || vms_by_uid_ems[h[:uid_ems]])
```

Comment:
If you need an array you can change the original index creation to do group_by instead of index_by

Reply:
:+1:

Reply:
Just be careful of your `&.` change below (just noticed)